src/pages/library/DataBrowser.js

```diff
@@ -33,117 +82,181 @@ const sidebarSections = [{
-      },
-      description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.',
-      lowerName: 'this is a really really long name that should wrap onto another line',
-      lowerDescription: 'test desc',
-      project: {
-        id: '112',
-        name: 'NIH'
-      },
-      subjects: 123,
-      dataType: 'RNA Seq',
-      keepCollapsed: true,
-      locked: true,
-      files: 45,
-      fileSize: 12345
-    },
-    {
-      namespace: 'harry-potter',
-      name: 'Harry Potter',
-      created: '2020-01-07T18:25:28.340Z',
-      lastUpdated: '2020-01-07T18:25:28.340Z',
-      tags: {
-        itemsType: 'AttributeValue',
-        items: ['1000 Genomes', 'CMG', 'Open Access', 'Exome']
-      },
-      description: 'The boy who lived',
-      lowerName: 'harry potter',
-      lowerDescription: 'lower description',
-      project: {
-        id: '112',
-        name: 'CDC'
-      },
-      subjects: 10,
-      dataType: 'RNA Seq',
-      keepCollapsed: true,
-      locked: false,
-      files: 15,
-      fileSize: 4432
-    },
-    {
-      namespace: 'star-wars',
-      name: 'Luke Skywalker',
-      created: '2020-01-07T18:25:28.340Z',
-      lastUpdated: '2020-01-07T18:25:28.340Z',
-      tags: {
-        itemsType: 'AttributeValue',
-        items: ['1000 Genomes', 'CMG', 'Open Access', 'Exome']
-      },
-      description: 'force sensitive',
-      lowerName: 'luke skywalker',
-      lowerDescription: 'lower description',
-      project: {
-        id: '123',
-        name: 'CDC'
-      },
-      subjects: 10,
-      dataType: '',
-      keepCollapsed: true,
-      locked: false,
-      files: 10,
-      fileSize: 550
-    },
-    {
-      namespace: 'star-wars',
-      name: 'Darth Vader',
-      created: '2020-01-07T18:25:28.340Z',
-      lastUpdated: '2020-01-07T18:25:28.340Z',
-      tags: {
-        itemsType: 'AttributeValue',
-        items: ['1000 Genomes', 'CCDG', 'Controlled Access', 'Whole Genome', 'asthma']
-      },
-      description: 'force sensitive',
-      lowerName: 'darth vader',
-      lowerDescription: 'lower description',
-      project: {
-        id: '123',
-        name: 'CDC2'
-      },
-      subjects: 10,
-      dataType: 'Data',
-      keepCollapsed: true,
-      locked: true,
-      files: 10,
-      fileSize: 2200000000
+const extractTags = snapshot => {
+  return {
+    itemsType: 'AttributeValue',
+    items: [
+      snapshot.locked ? 'controlled access' : 'open access',
+      _.toLower(snapshot.project),
+      ..._.map('dcat:mediaType', snapshot.files)
+    ]
+  }
+}
+
+const Browser = () => {
```

Comment:
As a rule, we try not to declare components inside of other components, as that results in them being redeclared every time the component is re-evaluated, which prevents React from being able to understand that they're the same component.

Could you move `SelectedItemDisplay` and `DataTable`, so they're declared outside of `Browser`, if we're not in-lining them? I'd also recommend renaming `DataTable` to something more specific, as we already have a major component with that name. The signatures would be:
```js
const SelectedItemsDisplay = ({ selectedData, setSelectedData }) => {}
const DataBrowserTable = ({ listData, sort, setSort, selectedData, toggleSelectedData, setRequestDatasetAccessList }) => {}
```

Reply:
gotcha, nice catch 👍